### PR TITLE
Topic/show inbuilt functions

### PR DESCRIPTION
**Pull request by HuwCampbell**

### CODE REVIEW

File: src/Icicle/Source/Query/Exp.hs [hunk 1]
```diff
@@ -93,7 +93,7 @@ data Fun
  | DaysBetween
  | DaysEpoch
  | Seq
- deriving (Show, Eq, Ord)
+ deriving (Show, Eq, Ord, Enum, Bounded)
```
Comment (amosr) [inline src/Icicle/Source/Query/Exp.hs]: can you make a new function here, ``listOfAllPrims :: [Prim]``? I don't mind the enum/bounded trick but I'd rather it be very localised. that way if we decide to add more interesting, non-enumerable prims, we only need to modify this file

File: main/icicle-repl.hs [hunk 2]
```diff
@@ -530,8 +531,8 @@ showState state
     , flag "c-eval:       " hasSeaEval
     ,      "now:          " <> T.unpack (renderTime $ currentTime state)
     ,      "data:         " <> show (length $ facts state) <> " rows"
-    ,      "dictionary:   " <> show (prettyDictionarySummary (dictionary state))
```
Comment (tranma) [inline main/icicle-repl.hs]: hm, why is this removed?

File: main/icicle-repl.hs [hunk 3]
```diff
@@ -530,8 +531,8 @@ showState state
...
     ]
+ prettyHL $ prettyDictionarySummary (dictionary state)
```
Comment (tranma) [inline main/icicle-repl.hs]: oh, you just print it separately

**Comment by tranma** [comment]
I like the fresh namer hack :+1:

File: src/Icicle/Source/Query/Exp.hs (hunk 1 above)
Comment (HuwCampbell) [inline src/Icicle/Source/Query/Exp.hs]: Sure

**Comment by amosr** [comment]
:+1: if you want to merge this in that'd be cool, if you don't, I'll get around to it after looking at the core rejig
